library/math/count_paths/count_paths_rectangle.hpp

```diff
@@ -0,0 +1,37 @@
+/** @file */
+#pragma once
+#include "../../../kactl/content/numerical/NumberTheoreticTransform.h"
+#include "../n_choose_k/n_choose_k.hpp"
+/**
+ * @see https://noshi91.hatenablog.com/entry/2023/07/21/235339
+ * @code{.cpp}
+       auto [right, top] = get_right_and_top(left, bottom);
+ * @endcode
+ * @param left,bottom see image
+ * @returns [right, top] -> see image, note right.back() == top.back()
+ * @time O((n + m)log(n + m))
+ * @space O(n + m)
+ */
+array<vl, 2> get_right_and_top(vl left, vl bottom) {
```

Comment:
in theory we could pass by reference I think... since they get swapped back

if we wanted to make them const we could create pointers to them which we swap instead

Reply:
if we do non-const reference, then we have to call it like:

vl r(some iterators);
get_right_and_top(r...);

and u do a copy eithet way

Reply:
like the only way i can think of which avoids a copy is if `get_right_and_top` takes in 2 pairs of iterators (representing subarrays of left,bottom)

Reply:
ugh in rust this would be so clean (and probably less copies)